examples/minwebgl/filter/src/main.rs

```diff
@@ -0,0 +1,77 @@
+use minwebgl as gl;
+use gl::GL;
+use web_sys::
+{
+  wasm_bindgen,
+  HtmlCanvasElement,
+  HtmlImageElement,
+};
+use wasm_bindgen::prelude::*;
+
+fn main()
+{
+  gl::browser::setup( Default::default() );
+  run();
+}
+
+fn run()
+{
+  let image_path = "static/download.png";
+  let gl = gl::context::retrieve_or_make().expect( "Can't retrieve GL context" );
+
+  let on_load = move | img : &HtmlImageElement |
+  {
+    let texture = gl.create_texture();
+    gl.bind_texture( GL::TEXTURE_2D, texture.as_ref() );
+    gl.tex_parameteri( GL::TEXTURE_2D, GL::TEXTURE_WRAP_S, GL::CLAMP_TO_EDGE as i32 );
+    gl.tex_parameteri( GL::TEXTURE_2D, GL::TEXTURE_WRAP_T, GL::CLAMP_TO_EDGE as i32 );
+
+    gl.pixel_storei( GL::UNPACK_FLIP_Y_WEBGL, 1 );
+    gl.tex_image_2d_with_u32_and_u32_and_html_image_element
+    (
+      GL::TEXTURE_2D,
+      0,
+      GL::RGBA as i32,
+      GL::RGBA,
+      GL::UNSIGNED_BYTE,
+      &img,
+    ).expect( "Can't load an image" );
+    gl.pixel_storei( GL::UNPACK_FLIP_Y_WEBGL, 0 );
+    gl.generate_mipmap( GL::TEXTURE_2D );
+
+    let canvas = gl.canvas().expect( "Canvas should exist" ).dyn_into::< HtmlCanvasElement >().unwrap();
+    canvas.set_width( img.width() );
+    canvas.set_height( img.height() );
+    gl.viewport( 0, 0, img.width() as i32, img.height() as i32 );
+
+    let vertex_source = include_str!( "shaders/main.vert" );
+    let fragment_source = include_str!( "shaders/main.frag" );
+    let program = gl::ProgramFromSources::new( vertex_source, fragment_source )
```

Comment:
You compile shader program every time when image is loaded? Argue your solution, when it can be useful.